src/ui/common/src/components/pages/workflows/components/CheckItem.tsx

```diff
@@ -0,0 +1,142 @@
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { Box, Tooltip, Typography } from '@mui/material';
+import React from 'react';
+import { useState } from 'react';
+
+import { theme } from '../../../../styles/theme/theme';
+import { CheckLevel } from '../../../../utils/operators';
+import ExecutionStatus from '../../../../utils/shared';
+import { ShowMore } from './MetricItem';
+
+const errorIcon = (
+  <Tooltip title="Error" placement="bottom" arrow>
+    <Box sx={{ fontSize: '20px', color: theme.palette.red['500'] }}>
+      <FontAwesomeIcon icon={faCircleExclamation} />
+    </Box>
+  </Tooltip>
+);
+
+const warningIcon = (
+  <Tooltip title="Warning" placement="bottom" arrow>
+    <Box sx={{ fontSize: '20px', color: theme.palette.orange['500'] }}>
+      <FontAwesomeIcon icon={faTriangleExclamation} />
+    </Box>
+  </Tooltip>
+);
+
+const successIcon = (
+  <Tooltip title="Success" placement="bottom" arrow>
+    <Box sx={{ fontSize: '20px', color: theme.palette.green['400'] }}>
+      <FontAwesomeIcon icon={faCircleCheck} />
+    </Box>
+  </Tooltip>
+);
+
+const unknownIcon = (
+  <Tooltip title="Unknown" placement="bottom" arrow>
+    <Box sx={{ fontSize: '20px', color: theme.palette.gray['400'] }}>
+      <FontAwesomeIcon icon={faQuestionCircle} />
+    </Box>
+  </Tooltip>
+);
+
+const canceledIcon = (
+  <Tooltip title="Canceled" placement="bottom" arrow>
+    <Box sx={{ fontSize: '20px', color: theme.palette.gray['400'] }}>
+      <FontAwesomeIcon icon={faCircleXmark} />
+    </Box>
+  </Tooltip>
+);
+
+export interface CheckPreview {
+  checkId: string;
+  name: string;
+  status: ExecutionStatus;
+  level: CheckLevel;
+  value?: string;
+  // a date.toLocaleString() should go here.
+  timestamp: string;
+}
+
+interface CheckItemProps {
+  checks: CheckPreview[];
+}
+
+export const CheckItem: React.FC<CheckItemProps> = ({ checks }) => {
+  const [expanded, setExpanded] = useState(false);
+  const checksList = [];
+  let checksToShow = checks.length;
+
+  if (checks.length > 0) {
+    if (!expanded) {
+      checksToShow = 1;
+    }
+
+    for (let i = 0; i < checksToShow; i++) {
+      let statusIcon = successIcon;
+      console.log('checks[i]: ', checks[i]);
+      if (checks[i].status === ExecutionStatus.Failed) {
+        statusIcon = errorIcon;
+      } else if (checks[i].status === ExecutionStatus.Succeeded) {
+        // now we check the value to see if we should show warning or error icon
+        if (checks[i].value === 'False') {
+          if (checks[i].level === CheckLevel.Error) {
+            statusIcon = errorIcon;
+          } else {
+            statusIcon = warningIcon;
+          }
+        }
+      } else if (checks[i].status === ExecutionStatus.Canceled) {
+        statusIcon = canceledIcon;
+      } else if (checks[i].status !== ExecutionStatus.Succeeded) {
+        console.log('checks[i].status: ', checks[i].status);
+        statusIcon = unknownIcon;
+      }
```

Comment:
what would happen if non of the above `if` statements get hit?

Reply:
should we have some default behavior here?

Reply:
it would stay as succeeded.

i'm going to refactor this block and make it into a switch statement.

Reply:
default case will be unknown icon though